host_pool.js

```diff
@@ -0,0 +1,174 @@
+"use strict";
+
+var net = require('net');
+
+/* HostPool:
+ *
+ * Starts with a pool of backend hosts from a "forwarding_host_pool"
+ * configuration that looks like this:
+ *
+ *      1.1.1.1:11, 2.2.2.2:22, 3.3.3.3:33
+ *
+ * It randomizes the list and then gives then out sequentially (for
+ * predictability).
+ *
+ * If failed() is called with one of the hosts, we mark it down for retry_secs
+ * and don't give it out again until that period has passed.
+ *
+ * If *all* the hosts have been marked down, we ignore the marks and just give
+ * out the next host. That's too keep some random short-lived but widespread
+ * network failure from taking the whole system down.
+ */
+
+var logger = require('./logger');
+var utils  = require('./utils');
+
+// takes a comma/space-separated list if ip:ports
+//  1.1.1.1:22,  3.3.3.3:44
+function HostPool(hostports_str, retry_secs){
+    var self = this;
+
+    var hosts = (hostports_str || '')
+            .trim()
+            .split(/[\s,]+/)
+            .map(function(hostport){
+                var splithost = hostport.split(/:/);
+                return { host: splithost[0],
+                          port: splithost[1]
+                        };
```

Comment:
Since Haraka is a MTA, having a default port 25 (when unspecified) makes the config syntax a little more friendly.